[llvm/IR] Fix module build issue following e57308b063bb (NFC)

medismailben opened the pull request:
This patch fixes a build issue following e57308b063bb when enabling
module build.

With that change, we failed to build the LLVM_IR module since
GEPNoWrapFlags wasn't defined prior to using it.

This patch addressed that issue by including the missing header in
`llvm/IR/IRBuilderFolder.h` which uses the `GEPNoWrapFlags` type.
This should ensure that we can always build the `LLVM_IR` module.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>

nikic reviewed (changes requested):
This is not the right place for the include, as Constant.h does not even use GEPNoWrapFlags.

In which header does the module build report a problem?

medismailben commented:
@nikic I went with the lowest common denominator (which was `Constant.h`) since I thought I'd need to add that include in a bunch of places, but it looks like it's only needed in `ConstantFolder.h` & `IRBuilderFolder.h` (still building ...)

I'll update this PR to only have it in the appropriate headers.

nikic commented on llvm/include/llvm/IR/ConstantFolder.h:
```diff
@@ -20,6 +20,7 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/IR/ConstantFold.h"
 #include "llvm/IR/Constants.h"
+#include "llvm/IR/GEPNoWrapFlags.h"
```
Is this include needed? It pulls in IRBuilderFolder.h on the line directly below, so should get it transitively.

medismailben committed:
[llvm/IR] Fix module build issue following e57308b063bb (NFC)

This patch fixes a build issue following e57308b063bb when enabling
module build.

With that change, we failed to build the LLVM_IR module since
GEPNoWrapFlags wasn't defined prior to using it.

This patch addressed that issue by including the missing header in
`llvm/IR/IRBuilderFolder.h` which uses the `GEPNoWrapFlags` type.
This should ensure that we can always build the `LLVM_IR` module.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>

nikic reviewed (approved):
LGTM, thanks!

I think technically that header only needs a forward declare, but anything actually using IRBuilderFolder.h would need the include anyway, so this is fine.
